packages/dev/core/src/Lights/volumetricSpotLight.ts

```diff
@@ -0,0 +1,80 @@
+import { SpotLight } from "./spotLight";
+import { Scene } from "core/scene";
+import { Axis } from "core/Maths/math.axis";
+import { CreateCylinder } from "core/Meshes/Builders/cylinderBuilder";
+import { Mesh } from "core/Meshes/mesh";
+import { ShaderMaterial } from "core/Materials/shaderMaterial";
+import "../Shaders/volumetricSpot.fragment";
+import "../Shaders/volumetricSpot.vertex";
+
+export interface IVolumetricSpotLight {
+    diameterTop: number;
+    diameterBottom: number;
+    rayLength?: number;
+    scene: Scene;
+}
+
+export class VolumetricSpotLight implements IVolumetricSpotLight {
+    diameterTop: number;
+    diameterBottom: number;
+    rayLength?: number;

```

Comment:
You should make getters/setters for these properties, and recreate the cone mesh when they are updated. Or make these properties protected/private if we don't want the user to change them after creation.